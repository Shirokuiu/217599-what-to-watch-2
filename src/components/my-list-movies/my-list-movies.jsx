import React from "react";
import PropTypes from "prop-types";
import {connect} from "react-redux";

import {MovieList} from "../movie-list/movie-list";

export const MyListMovies = (props) => {
  const {moviesFavorite} = props;

  return <section className="catalog">
    <h2 className="catalog__title visually-hidden">Catalog</h2>

    <MovieList movies={moviesFavorite} />
  </section>;
};

MyListMovies.propTypes = {
  moviesFavorite: PropTypes.arrayOf(PropTypes.shape({
    id: PropTypes.number,
    name: PropTypes.string,
    posterImage: PropTypes.string,
    previewImage: PropTypes.string,
    backgroundImage: PropTypes.string,
    backgroundColor: PropTypes.string,
    videoLink: PropTypes.string,
    previewVideoLink: PropTypes.string,
    description: PropTypes.string,
    rating: PropTypes.number,
    scoresCount: PropTypes.number,
    director: PropTypes.string,
    starrings: PropTypes.array.string,
    runTime: PropTypes.number,
    genre: PropTypes.string,
    released: PropTypes.number,
    isFavorite: PropTypes.bool,
  })).isRequired
};

const mapStateToProps = (state, ownProps) => Object.assign({}, ownProps, {
  moviesFavorite: state.moviesFavorite
});

export default connect(mapStateToProps)(MyListMovies);
