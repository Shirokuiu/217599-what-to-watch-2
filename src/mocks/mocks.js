export const films = [
  {
    img: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    title: `Fantastic Beasts`
  },
  {
    img: `img/bohemian-rhapsody.jpg`,
    title: `Bohemian Rhapsody`
  },
  {
    img: `img/macbeth.jpg`,
    title: `Macbeth`
  },
  {
    img: `img/aviator.jpg`,
    title: `Aviator`
  },
  {
    img: `img/we-need-to-talk-about-kevin.jpg`,
    title: `We need to talk about Kevin`
  }
];
