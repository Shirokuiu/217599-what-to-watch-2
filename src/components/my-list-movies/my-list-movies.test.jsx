import React from "react";
import Enzyme, {shallow} from "enzyme";
import toJson from "enzyme-to-json";
import Adapter from "enzyme-adapter-react-16";
import {BrowserRouter} from "react-router-dom";

import {MyListMovies} from "../my-list-movies/my-list-movies";

import {films} from "../../mocks/mocks";

Enzyme.configure({adapter: new Adapter()});

it(`Is my list movies rendered`, () => {
  const tree = shallow(<BrowserRouter>
    <MyListMovies
      moviesFavorite={films}
    />
  </BrowserRouter>);

  expect(toJson(tree)).toMatchSnapshot();
});
