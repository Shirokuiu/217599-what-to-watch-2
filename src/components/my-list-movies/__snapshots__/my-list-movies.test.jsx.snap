// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Is my list movies rendered 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "createHref": [Function],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "pathname": "/",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <MyListMovies
    moviesFavorite={
      Array [
        Object {
          "backgroundColor": "#ffffff",
          "backgroundImage": "img/the-grand-budapest-hotel-bg.jpg",
          "description": "In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege.",
          "director": "Wes Andreson",
          "genre": "Comedy",
          "id": 1,
          "isFavorite": false,
          "name": "The Grand Budapest Hotel",
          "posterImage": "img/the-grand-budapest-hotel-poster.jpg",
          "previewImage": "img/the-grand-budapest-hotel.jpg",
          "previewVideoLink": "https://some-link",
          "rating": 8.9,
          "released": 2014,
          "runTime": 99,
          "scoresCount": 240,
          "starring": Array [
            "Bill Murray",
            "Edward Norton",
            "Jude Law",
            "Willem Dafoe",
            "Saoirse Ronan",
          ],
          "videoLink": "https://some-link",
        },
      ]
    }
  />
</Router>
`;
